daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -77,7 +86,16 @@ object SExpr {
   }
 
   /** Reference to a builtin function */
-  final case class SEBuiltin(b: SBuiltin) extends SExpr {
+  final case class SEBuiltin(b: SBuiltin) extends SExprAtomic {
+    def evaluate(machine: Machine): SValue = {
+      /* special case for nullary record constructors */
+      b match {
+        case SBRecCon(id, fields) if b.arity == 0 =>
+          SRecord(id, fields, new ArrayList())
```

Comment:
```suggestion
          SRecord(id, fields, new util.ArrayList())
```

Reply:
Done
The new code was a copy of some existing code in the same style, so I fixed that too.